cpp/src/arrow/compute/kernels/minmax.cc

```diff
@@ -71,6 +72,7 @@ struct MinMaxState<ArrowType, enable_if_floating_point<ArrowType>> {
 
   c_type min = std::numeric_limits<c_type>::infinity();
   c_type max = -std::numeric_limits<c_type>::infinity();
+  bool has_nulls = false;
```

Comment:
You also need to add logic handling this member to `operator+=` or behavior across multi chunked input will be inconsistent. (for example, `rhs.has_nulls == true` is currently ignored)

Reply:
@bkietz see the issue now, will fix it in next push, thanks!